### PR TITLE
Update README.md [skip ci]

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

- Removes outdated installation step
- Removes specific version mentions (so we don't have to remember to update it each release)
- Other minor changes

## Before submitting
- [n/a] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -369,8 +363,8 @@ class LitAutoEncoder(pl.LightningModule):
 ## Community
 
 The lightning community is maintained by
-- [16 core contributors](https://pytorch-lightning.readthedocs.io/en/latest/governance.html) who are all a mix of professional engineers, Research Scientists, Ph.D. students from top AI labs.
-- 280+ community contributors.
+- [16 core contributors](https://pytorch-lightning.readthedocs.io/en/latest/governance.html) who are all a mix of professional engineers, Research Scientists, and Ph.D. students from top AI labs.
+- 400+ community contributors.
```
Comment (carmocca) [inline README.md:367]: Currently at 397. Took the liberty of rounding up 😜

**Review by Borda — COMMENTED** [review]
This is duplicate to #5922

**Comment by carmocca** [comment]
@Borda changed base to your branch. No need to have both PRs. Feel free to merge

**Review by awaelchli — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ Lightning is rigurously tested across multiple GPUs, TPUs CPUs and against major
 </details>
 
 <details>
-  <summary>Bleeding edge build status (1.2)</summary>
+  <summary>Bleeding edge build status (future release)</summary>
```
Comment (carmocca) [inline README.md:95]:
```suggestion
  <summary>Bleeding edge build status (1.2)</summary>
```

**Pull request merged**